desktop/updater/index.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+/**
+ * External Dependencies
+ */
+const electron = require( 'electron' );
+const app = electron.app;
+
+/**
+ * Internal dependencies
+ */
+const platform = require( '../platform' );
+const config = require( '../config' );
+const AutoUpdater = require( './auto-updater' );
+const ManualUpdater = require( './manual-updater' );
+// const settings = require( 'lib/settings' );
+
+let updater = false;
+
+function urlBuilder( version ) {
+	let platformString = 'osx';
+
+	if ( platform.isWindows() ) {
+		platformString = 'windows';
+	} else if ( platform.isLinux() ) {
+		platformString = 'linux';
+	}
+
+	if ( config.forceUpdate ) {
+		version = '0.0.1';
+	}
+
+	return config.updater.url + platformString + '/version?compare=' + version;
+}
+
+module.exports = function() {
+	app.on( 'will-finish-launching', function() {
+		let url = urlBuilder( app.getVersion() );
+
+		if ( platform.isOSX() ) {
+			updater = new AutoUpdater( url );
+		} else {
+			updater = new ManualUpdater( url );
+		}
+
+		// Start one straight away
+		setTimeout( updater.ping.bind( updater ), config.updater.delay );
+		setInterval( updater.ping.bind( updater ), config.updater.interval );
```

Comment:
There's probably a valid reason to have the updater call itself or wrap this in a function that recursively calls `setTimeout` because `setInterval()` will attempt to call the given function every interval on the interval. Granted, the config is set at 10 minutes, but supposing there were network trouble or some freak delay or a misconfigured `config` maybe we wouldn't want to pile up those calls.